diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/BlockStructureBraces.kt

```diff
@@ -43,76 +52,130 @@ class BlockStructureBraces : Rule("block-structure") {
-        checkCloseBrace(node.findChildByType(BODY)!!.findChildByType(BLOCK)!!)
+    private fun checkLoop(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
+        if (node.hasChildOfType(BODY) && node.findChildByType(BODY)!!.hasChildOfType(BLOCK)) {
+            checkOpenBrace(node, BODY, configuration)
+            checkCloseBrace(node.findChildByType(BODY)?.findChildByType(BLOCK), configuration)
+        }
     }
 
-    private fun checkWhen(node: ASTNode) {
-        checkOpenBrace(node, LBRACE)
-        checkCloseBrace(node)
+    private fun checkWhen(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
+        checkOpenBrace(node, LBRACE, configuration)
+        checkCloseBrace(node, configuration)
     }
 
-    private fun checkFun(node: ASTNode) {
+    private fun checkFun(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
         if (node.hasChildOfType(BLOCK)) {
-            checkOpenBrace(node, BLOCK)
-            checkCloseBrace(node.findChildByType(BLOCK)!!)
+            checkOpenBrace(node, BLOCK, configuration)
+            checkCloseBrace(node.findChildByType(BLOCK)!!, configuration)
         }
     }
 
-    private fun checkIf(node: ASTNode) {
+    private fun checkIf(node: ASTNode, configuration: BlockStructureBracesConfiguration) {
         val ifPsi = node.psi as KtIfExpression
         val thenNode = ifPsi.then?.node
         val hasElseBranch = ifPsi.elseKeyword != null
         val elseNode = ifPsi.`else`?.node
-
-        checkOpenBrace(node, THEN)
-        checkCloseBrace(thenNode!!)
-        if (hasElseBranch && elseNode!!.elementType != IF) {
-            checkOpenBrace(node, ELSE)
-            checkCloseBrace(elseNode)
+        if (thenNode != null && thenNode.hasChildOfType(LBRACE)) {
+            checkOpenBrace(node, THEN, configuration)
+            checkCloseBrace(thenNode, configuration)
+        }
+        if (hasElseBranch && elseNode!!.elementType != IF && elseNode.hasChildOfType(LBRACE)) {
+            checkOpenBrace(node, ELSE, configuration)
+            checkCloseBrace(elseNode, configuration)
         }
     }
 
-    private fun checkOpenBrace(node: ASTNode, beforeType: IElementType) {
+    private fun checkOpenBrace(node: ASTNode, beforeType: IElementType, configuration: BlockStructureBracesConfiguration) {
+        if (!configuration.openBrace) return
         val braceSpace = node.findChildBefore(beforeType, WHITE_SPACE)
-        if (!(braceSpace == null || braceSpace.elementType != WHITE_SPACE)) {
+        if (braceSpace != null) {
             if (braceSpace.text.contains("\n".toRegex())) {
-                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "open braces", 0) {
+                BRACES_BLOCK_STRUCTURE_ERROR.warnAndFix(configRules, emitWarn, isFixMode, "incorrect newline before opening brace",
+                        (braceSpace.prevSibling {it.elementType == beforeType} ?:braceSpace).startOffset){
```

Comment:
`braceSpace` is a node before `beforeType`, isn't it? Why are you looking for `prevSibling` with `beforeType`?